ingest-start-workflow/src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -1,57 +1,41 @@
 package uk.gov.nationalarchives
 
 import cats.effect._
-import cats.effect.unsafe.implicits.global
-import com.amazonaws.services.lambda.runtime.{Context, RequestStreamHandler}
-import org.typelevel.log4cats.{LoggerName, SelfAwareStructuredLogger}
-import org.typelevel.log4cats.slf4j.Slf4jFactory
-import pureconfig.ConfigSource
+import io.circe.generic.auto._
 import pureconfig.generic.auto._
-import pureconfig.module.catseffect.syntax._
 import uk.gov.nationalarchives.Lambda._
 import uk.gov.nationalarchives.dp.client.WorkflowClient
 import uk.gov.nationalarchives.dp.client.WorkflowClient.{Parameter, StartWorkflowRequest}
 import uk.gov.nationalarchives.dp.client.fs2.Fs2Client
-import upickle.default
-import upickle.default._
 
-import java.io.{InputStream, OutputStream}
+class Lambda extends LambdaRunner[Input, StateOutput, Config, Dependencies] {
 
-class Lambda extends RequestStreamHandler {
-  private val configIo: IO[Config] = ConfigSource.default.loadF[IO, Config]()
-  implicit val loggerName: LoggerName = LoggerName(sys.env("AWS_LAMBDA_FUNCTION_NAME"))
-  private val logger: SelfAwareStructuredLogger[IO] = Slf4jFactory.create[IO].getLogger
-
-  lazy val workflowClientIO: IO[WorkflowClient[IO]] = configIo.flatMap { config =>
-    Fs2Client.workflowClient(config.apiUrl, config.secretName)
-  }
-
-  implicit val inputReader: Reader[Input] = macroR[Input]
-
-  override def handleRequest(inputStream: InputStream, output: OutputStream, context: Context): Unit = {
-    val inputString = inputStream.readAllBytes().map(_.toChar).mkString
-    val input = read[Input](inputString)
+  override def handler: (
+      Input,
+      Config,
+      Dependencies
+  ) => IO[StateOutput] = (input, _, dependencies) => {
     val batchRef = input.executionId.split("-").take(3).mkString("-")
-    val log = logger.info(Map("batchRef" -> batchRef))(_)
+    val logWithBatch = log(Map("batchRef" -> batchRef))(_)
     for {
-      _ <- log(s"Starting workflow ${input.workflowContextName} for $batchRef")
-      workflowClient <- workflowClientIO
-      id <- workflowClient.startWorkflow(
+      _ <- logWithBatch(s"Starting dependencies workflow ${input.workflowContextName} for $batchRef")
```

Comment:
Shouldn't this be "Starting *ingest* workflow..."?

Reply:
Good spot, I've changed it.